hnn_core/extracellular.py

```diff
@@ -0,0 +1,601 @@
+            traces. The SciPy function :func:`~scipy.signal.decimate` is used,
+            which recommends values <13. To achieve higher decimation factors,
+            a list of ints can be provided. These are applied successively.
+        color : string | array of floats | matplotlib.colors.ListedColormap
+            The color to use for plotting (optional). The usual Matplotlib
+            standard color strings may be used (e.g., 'b' for blue). A color
+            can also be defined as an RGBA-quadruplet, or an array of
+            RGBA-values (one for each electrode contact trace to plot). An
+            instance of :class:`~matplotlib.colors.ListedColormap` may also be
+            provided.
+        voltage_offset : float | None (optional)
+            Amount to offset traces by on the voltage-axis. Useful for plotting
+            laminar arrays.
+        voltage_scalebar : float | None (optional)
+            Height, in units of uV, of a scale bar to plot in the top-left
+            corner of the plot.
+        contact_labels : list (optional)
+            Labels associated with the contacts to plot. Passed as-is to
+            :meth:`~matplotlib.axes.Axes.set_yticklabels`.
+        show : bool
+            If True, show the figure
+
+        Returns
+        -------
+        fig : instance of plt.fig
+            The matplotlib figure handle into which time series were plotted.
+        """
+        from .viz import plot_extracellular
+
+        if trial_no is None:
+            plot_data = self.get_data()
+        elif isinstance(trial_no, (list, tuple, int, slice)):
+            plot_data = self.get_data()[trial_no, ]
+        else:
+            raise ValueError(f'unkown trial number type, got {trial_no}')
+
+        if isinstance(contact_no, (list, tuple, int, slice)):
+            plot_data = plot_data[:, contact_no, ]
+        elif contact_no is not None:
+            raise ValueError(f'unkown contact number type, got {contact_no}')
+
+        for trial_data in plot_data:
+            fig = plot_extracellular(
+                self.times, trial_data, tmin=tmin, tmax=tmax, ax=ax,
+                decim=decim, color=color,
+                voltage_offset=voltage_offset,
+                voltage_scalebar=voltage_scalebar,
+                contact_labels=contact_labels,
+                show=show)
+        return fig
+
+
+class ExtracellularArrayBuilder(object):
+    """The ExtracellularArrayBuilder class
+
+    Parameters
+    ----------
+    array : ExtracellularArray object
+        The instance of :class:`hnn_core.extracellular.ExtracellularArray` to
+        build in NEURON-Python
+    """
+    def __init__(self, array):
+        self.array = array
+        self.n_contacts = array.n_contacts
+        self._nrn_imem_ptrvec = None
+        self._nrn_imem_vec = None
+        self._nrn_r_transfer = None
+        self._nrn_times = None
+        self._nrn_voltages = None
```

Comment:
like the `_nrn_xxxx` convention! Good variable names are half the battle won in programming